pkg/restapi/operation/operations.go

```diff
@@ -58,6 +59,17 @@ type Handler interface {
 	Handle() http.HandlerFunc
 }
 
+// UUIDBuilder represents a type that can generate a uuid string
+type UUIDBuilder interface {
```

Comment:
@DRK3 @grace0925 what are the reasons for this abstraction? If we want to plug in different Vault ID generators then it should be named differently... like `type VaultID func() string`.

Otherwise feel free to ditch this interface and just use `uuid.New().String()` instead.

Reply:
@llorllale I did this in order to replicate the error where uuid.new() generates a vault name that already exists for unit tests. Since the vault names are now randomly generated during creation, this is the only way I can think to to replicate this error... I will update the comments so it's less confusing.

Reply:
@grace0925 that's a bit "out there" but ok....

I would personally not bother.

`uuid.New()` uses [`uuid.NewRandom()`](https://pkg.go.dev/github.com/google/uuid#NewRandom):

> Randomly generated UUIDs have 122 random bits.  One's annual risk of being
hit by a meteorite is estimated to be one chance in 17 billion, that
means the probability is about 0.00000000006 (6 × 10−11),
equivalent to the odds of creating a few tens of trillions of UUIDs in a
year and having one duplicate.

But if the code is there already then carry on... 

Please, do put a comment in there explaining the reasoning behind the interface. :)

Reply:
@llorllale It's there in order to test the "duplicate vault" scenario. That being said... @grace0925 perhaps you can see how the unit test coverage looks without that case covered... alternatively there might be another way to test the "duplicate vault" case... (I'll message you about this later)

Reply:
@llorllale @grace0925 Once you add the check to prevent duplicate referenceIDs (similar to the Transmute implementation), that would be a more realistic duplicate vault scenario.

Reply:
@grace0925 After you do that, take a look again and see if you still need this mock UUID